testing/test_deploy_app.py

```diff
@@ -101,13 +130,52 @@ def insert_ssl_cert(args):
     shutil.rmtree(SSL_DIR)
   os.mkdir(SSL_DIR)
   logging.info("donwload ssl cert and insert to GKE cluster")
-  os.system("gsutil cp gs://%s/%s/* %s" % (SSL_BUCKET, args.cert_group, SSL_DIR))
-  os.system("gcloud container clusters get-credentials %s --zone %s --project %s" % (args.deployment, args.zone, args.project))
-  os.system("kubectl create -f %s" % SSL_DIR)
+  try:
+    util_run(("gsutil cp gs://%s/%s/* %s" % (SSL_BUCKET, args.mode, SSL_DIR)).split(' '))
+  except Exception:
+    logging.warning("ssl cert for %s doesn't exist in gcs" % args.mode)
+    return
+  util_run(("gcloud container clusters get-credentials %s --zone %s --project %s" %
+            (args.deployment, args.zone, args.project)).split(' '))
+  util_run(("kubectl create -f %s" % SSL_DIR).split(' '))
 
 
 def check_deploy_status(args):
   logging.info("check deployment status")
+  service_account_credentials = get_service_account_credentials("CLIENT_ID")
+
+  google_open_id_connect_token = get_google_open_id_connect_token(
+    service_account_credentials)
+  # Wait up to 30 minutes for IAP access test.
+  retry_credit = 180
+  status_code = 0
+  while retry_credit > 0:
```

Comment:
Why not just use the retrying module?

Reply:
If your doing time based retrying then I would suggest doing something like

end_time = datetime.datetime.now() + datetime.timedelta(minutes=30)
while datetime.datetime.now() < end_time:
...

This way if you change the sleep; you don't affect the total time we wait.